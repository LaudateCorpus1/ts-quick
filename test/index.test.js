const path = require('path');
const execa = require('execa');

process.chdir(__dirname);

const main = (args, options) => execa(path.join(__dirname, '../cli-main.js'), args, options);

test('ignore files in .gitignore', async () => {
  // TODO: Figure out why the spy on write isn't being called
  // const write = jest.spyOn(process.stdout, 'write');
  const cwd = path.join(__dirname, 'fixtures/gitignore');

  let err;
  try {
    await main([], {cwd});
  } catch(error) {
    err = error
  }

  expect(err).toMatchSnapshot();
});
