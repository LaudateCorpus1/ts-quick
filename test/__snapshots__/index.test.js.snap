// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignore files in .gitignore 1`] = `
[Error: Command failed: /Users/kushnem/dev/personal/ts-quick/cli-main.js

[96mbar.js[0m:[93m1[0m:[93m17[0m - [91merror[0m[90m TS7006: [0mParameter 'n' implicitly has an 'any' type.

[7m1[0m function square(n) {
[7m [0m [91m                ~[0m
]
`;
